tests/test_diaCalculationPlugins.py

```diff
@@ -872,6 +880,76 @@ def testCalculate(self):
                                np.sqrt(1 / (n_sources - 1)))
 
 
+class TestMultiLombScarglePeriodogram(unittest.TestCase):
```

Comment:
I think you need to add cases for N<4, etc., to ensure the code properly handles various potential failure modes.

Reply:
Yes this has been added at lines 938-940.